File: HLC01_03Java/src/main/java/org/example/juegoRol03/GamblingSpell.java
package org.example.juegoRol03;

import java.util.Random;

public class GamblingSpell implements Spell {
    private final String name = "ludopatía";

    public String getName() {
        return name;
    }

    private boolean isPair(int num) {
        return num % 2 == 0;
    }

    @Override
    public void cast(Personaje caster, Personaje target) {
        Random random = new Random();
        int outcome = random.nextInt(100) + 1; // Random between 1 and 100 (inclusive)

        if (isPair(outcome)) {
            System.out.println("Buena suerte");
            target.receiveDamage(999999999); // Apply massive damage
        } else {
            System.out.println("Mala suerte");
            caster.receiveDamage(999999999); // Apply massive damage
        }
    }

}

File: HLC01_03Kotlin/src/main/kotlin/juegoRol03/GamblingSpell.kt
package org.example.juegoRol03

import kotlin.random.Random

class GamblingSpell : Spell {
    override val name: String = "ludopatía"


    private fun isPair(num: Int): Boolean {
        return num % 2 == 0
    }


    override fun cast(caster: Personaje, target: Personaje) {
        val outcome = Random.nextInt(1, 101)
        if (isPair(outcome)) {
            println("Buena suerte")
            target.receiveDamage(9999999)
        } else {
            println("Mala suerte")
            caster.receiveDamage(9999999)
        }
    }

}